analysis_runner/cli_analysisrunner.py

```diff
@@ -277,3 +251,99 @@ def _perform_shebang_check(script: str) -> None:
                 f', consider inserting "{suggestion_shebang}" at the top of this file'
             )
         raise AssertionError(message)
+
+
+def get_repository_specific_information(  # noqa: C901
+    repository: str | None,
+    commit: str | None,
+    branch: str | None,
+    cwd: str | None,
+    script: list[str],
+) -> RepositorySpecificInformation:
+    """Do all the repository specific stuff (like fetching repo, commit, cwd)
+
+    Args:
+        repository (str | None): _description_
+        commit (str | None): _description_
+        branch (str | None): _description_
+        cwd (str | None): _description_
+        script (list[str]): _description_
+
+    Raises:
+        ValueError: _description_
+        SystemExit: _description_
+        SystemExit: _description_
+
+    Returns:
+        RepositorySpecificInformation: _description_
+    """
```

Comment:
Should the descriptions be changed here to something else? Variables to me seem pretty self explanatory but may as well?

Reply:
Yeah agreed, the auto docstring generator wasn't much use, and doesn't add a lot of information